### PR TITLE
Adds some sanity checks on the model definition

### PR DESCRIPTION
**Pull request by msaisumanth**

### CODE REVIEW

File: ludwig/utils/defaults.py [hunk 1]
```diff
@@ -130,7 +130,40 @@ def get_default_optimizer_params(optimizer_type):
         raise ValueError('Incorrect optimizer type: ' + optimizer_type)
 
 
+def _perform_sanity_checks(model_definition):
+    assert 'input_features' in model_definition, "Model definition doesn't " \
```
Comment (ydudin3) [inline ludwig/utils/defaults.py]: Why using double quotes?

File: ludwig/utils/defaults.py [hunk 2]
```diff
@@ -130,7 +130,40 @@ def get_default_optimizer_params(optimizer_type):
...
+                                                 "define any input features."
+
+    assert 'output_features' in model_definition, "Model definition doesn't " \
+                                                  "define any output features."
+
+    assert isinstance(model_definition['input_features'], list), \
```
Comment (ydudin3) [inline ludwig/utils/defaults.py]: break arguments into multiple lines, e.g.
isinstance(
    model_definition,
    ...
)

File: ludwig/utils/defaults.py [hunk 3]
```diff
@@ -130,7 +130,40 @@ def get_default_optimizer_params(optimizer_type):
...
+        "Ludwig expects input features in a list. Check your model " \
+        "definition format"
+
+    assert isinstance(model_definition['output_features'], list), \
+        "Ludwig expects output features in a list. Check your model " \
```
Comment (ydudin3) [inline ludwig/utils/defaults.py]: Check PEP8 guidelines on usage of `\` to break long lines

Comment (msaisumanth) [reply ludwig/utils/defaults.py]: Will update to use ()

File: ludwig/utils/defaults.py [hunk 4]
```diff
@@ -130,7 +130,55 @@ def get_default_optimizer_params(optimizer_type):
...
+        'Model definition needs to have at least one input feature'
+    )
+
+    assert len(model_definition['output_features']) > 0, (
+        'Model definition needs to have at least one output feature'
+    )
+
+    if 'training' in model_definition:
+        assert isinstance(model_definition['training'], dict), (
+            'There is an issue while reading the training section of the '
```
Comment (w4nderlust) [inline ludwig/utils/defaults.py]: Please specify here and in the next 2 messages that the value should be a dictionary

**Pull request closed without merging**